certbot/cert_manager.py

```diff
@@ -139,6 +149,90 @@ def update_certs_for_domain_matches(candidate_lineage, rv):
 
     return _search_lineages(config, update_certs_for_domain_matches, (None, None))
 
+def cert_path_to_lineage(config):
+    """ If config.cert_path is defined, find an appropriate value for config.certname
+    by searching through available files in config.renewal_configs_dir, and finding
+    one with an appropriate value for 'fullchain'."""
+
+    # pylint: disable=unused-argument
+    def update_cert_name_for_cert_path_match(candidate_lineage, rv):
+        """ Return the lineagename or return None. """
+        if rv:
```

Comment:
If instead of using these two lines, you use the append-to-a-list that `overlapping_archive_dirs` currently does, then we can remove that separate function and just have it inline here, which will additionally check for any type of multiple matching at all.